tests/e2e/e2e_setup_test.go

```diff
@@ -785,6 +787,29 @@ func (s *IntegrationTestSuite) writeExpeditedSoftwareUpgradeProp(c *chain) {
 	s.Require().NoError(err)
 }
 
+func (s *IntegrationTestSuite) writeERC20RegisterProposal(c *chain, erc20Address common.Address) {
```

Comment:
Maybe we could centralize these specific functions by its own files and tests.
For example, we would move this to the ERC20 testing file.
Also missing docs